pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotTableRestletResource.java

```diff
@@ -210,15 +213,36 @@ public String listTableConfigs(@ApiParam(value = "realtime|offline") @QueryParam
 
       if (tableType == null) {
         tableNames = _pinotHelixResourceManager.getAllRawTables();
+      } else if (tableType == TableType.REALTIME) {
+        tableNames = _pinotHelixResourceManager.getAllRealtimeTables();
       } else {
-        if (tableType == TableType.REALTIME) {
-          tableNames = _pinotHelixResourceManager.getAllRealtimeTables();
-        } else {
-          tableNames = _pinotHelixResourceManager.getAllOfflineTables();
-        }
+        tableNames = _pinotHelixResourceManager.getAllOfflineTables();
       }
 
-      Collections.sort(tableNames);
+      if (sortOrderByTime == "none") {
+        // Sort table names alphabetically
+        Collections.sort(tableNames);
+      } else {
+        // Sort table names based on (1) Create Time (2) Last Modified Time
+        TableType finalTableType = tableType;
+        Collections.sort(tableNames, (Comparator<String>) (o1, o2) -> {
+          TableConfig t1, t2;
+          if (finalTableType == TableType.REALTIME) {
+            t1 = _pinotHelixResourceManager.getRealtimeTableConfig(o1);
```

Comment:
The time info is already lost when converting `ZNRecord` to `TableConfig`. You may refer to `PinotHelixResourceManager.getTableStats()` on how to retrieve the `ZNRecord`

Reply:
Sure, let me take a look. Thanks for pointing out.